### ISSUE TITLE
SAML Drupal module configuration and testing

### ISSUE THREAD

**Issue by malikkotob** [18F/beta.foia.gov#112]
As an agency user, in order to login to update agency and agency component contact information, I need the Drupal backend to authenticate with Max.gov.

Once we have successfully configured SAML authentication between Drupal and the Max.gov IdP (#95), we are ready to configure the `simplesamlphp_auth` module.

## Acceptance criteria

- [x] Module configuration is completed as per https://blt.readthedocs.io/en/8.x/readme/simplesamlphp-setup/#module-configuration
- [x] Authentication with Max.gov IdP is working via Drupal login

## Developer Review

- [x] Review [PR 84](https://github.com/usdoj/foia-api/pull/84)

**Setup**
- [x] Deploy `doj-112-mk` to the dev environment
- [x] Import config: `drush cim vcs`
- [x] Clear caches: `drush cr`
- [x] Navigate to http://foiadev.prod.acquia-sites.com/simplesaml/module.php/core/frontpage_config.php
- [x] Verify that the three links under the configuration section don't show any errors (note: you will have to login as an admin, contact Malik for the pw)
- [x] Follow functional review steps

## Functional Review
- [x] Follow the steps at https://test.max.gov/password to set a password in the max.gov test environment at https://test.max.gov/password
- [x] Login to the dev environment using drush: `drush uli`
- [x] Go to `/admin/people/create`
- [x] Enter the e-mail address associated to your Max.gov account in the e-mail address field
- [x] Enter the same e-mail address in the username field
- [x] Under roles, check the `Agency Manager` role
- [x] For the Agency field, enter Department of Justice
- [x] Click 'Create new Account'
- [x] Logout
- [x] Go to `/user/login`
- [x] Click the "Login via MAX.gov" link
- [x] Verify you are redirected to the MAX.gov website to enter your credentials (**Note: ** If you are already logged in at MAX.gov you should just be logged right into Drupal)
- [x] Enter your Max username and password and Login
- [x] Verify you are redirected back to the Drupal site and are logged in
- [x] Click Logout
- [x] Verify you are logged out

**Comment by malikkotob** [comment]
Will start this and advance it as far as I can.

**Comment by malikkotob** [comment]
Unblocked and ready for review.

**Comment by ba66e77** [comment]
Behavior works as expected.  I did get a warning page after authenticating to max.gov, just before being redirected back.  I presume this is because we are testing on the acquia-sites domain and this will not be the case when testing on the fully provisioned domain with verified SSL cert.

<img width="1028" alt="mozilla_firefox" src="https://user-images.githubusercontent.com/1698983/31364730-ad1b33ac-ad34-11e7-8fc8-7325a0477873.png">

**Comment by dangur** [comment]
Functional review pass. I did not get that security warning ^^.

**Comment by malikkotob** [comment]
Maybe a difference in browsers. I think you're right @ba66e77 in that this shouldn't be the case when testing on the fully provisioned domain.

Merged into `develop`.
